sdcm/kafka/kafka_producer.py

```diff
@@ -0,0 +1,410 @@
+                self.schema_registry_client, self.key_schema, lambda obj, ctx: obj,
+            ) if self.key_schema else None
+            self.value_serializer = AvroSerializer(
+                self.schema_registry_client, self.value_schema, lambda obj, ctx: obj,
+            ) if self.value_schema else None
+
+            self.key_deserializer = AvroDeserializer(self.schema_registry_client)
+            self.value_deserializer = AvroDeserializer(self.schema_registry_client)
+        else:
+            self.key_serializer = StringSerializer('utf_8')
+            self.value_serializer = StringSerializer('utf_8')
+            self.key_deserializer = StringDeserializer('utf_8')
+            self.value_deserializer = StringDeserializer('utf_8')
+
+    def generate_data_from_schema(self, schema: dict, seed: int) -> dict | int | float | str | bool | bytes | None:
+        rand_gen = random.Random(seed)
+        if schema['type'] in ('record', 'struct'):  # 'record' type of schema for AVRO record format, 'struct' for JSON
+            data = {}
+            for field in schema['fields']:
+                # 'name' is used in AVRO type records, 'field' in JSON ones
+                field_name = field.get('name') or field.get('field')
+                if field_name == 'id':
+                    data[field_name] = seed
+                else:
+                    data[field_name] = self.generate_field_value(field['type'], rand_gen)
+            return data
+        else:
+            return self.generate_field_value(schema['type'], rand_gen)
+
+    def generate_field_value(self, field_type: str, rand_gen: random.Random) -> int | float | str | bool | bytes | None:
+        type_mapping = {
+            'int8': 'int',
+            'int16': 'int',
+            'int32': 'int',
+            'int64': 'long',
+            'int': 'int',
+            'long': 'long',
+            'float32': 'float',
+            'float64': 'double',
+            'float': 'float',
+            'double': 'double',
+            'boolean': 'boolean',
+            'string': 'string',
+        }
+        normalized_type = type_mapping.get(field_type, field_type)
+
+        value = None
+        if normalized_type == 'int':
+            value = rand_gen.randint(0, 1000000)
+        elif normalized_type == 'long':
+            value = rand_gen.randint(0, 10000000000)
+        elif normalized_type == 'string':
+            value = ''.join(rand_gen.choices(ascii_letters, k=10))
+        elif normalized_type == 'boolean':
+            value = rand_gen.choice([True, False])
+        elif normalized_type == 'float':
+            value = rand_gen.uniform(0, 1000000)
+        elif normalized_type == 'double':
+            value = rand_gen.uniform(0, 10000000000)
+        elif normalized_type == 'bytes':
+            value = ''.join(rand_gen.choices(ascii_letters, k=10)).encode('utf-8')
+        else:
+            LOGGER.warning("Unsupported field type: %s. Using None.", field_type)
```

Comment:
should raise error event or fail the test immediately

Reply:
Added raising error event.